src/controlCenter/controlCenter.vala

```diff
@@ -147,39 +148,36 @@ namespace SwayNotificationCenter {
-                                              false);
-                    GtkLayerShell.set_anchor (this,
-                                              GtkLayerShell.Edge.LEFT,
-                                              true);
-                    break;
-                case PositionX.CENTER:
-                    GtkLayerShell.set_anchor (this,
-                                              GtkLayerShell.Edge.RIGHT,
-                                              false);
-                    GtkLayerShell.set_anchor (this,
-                                              GtkLayerShell.Edge.LEFT,
-                                              false);
-                    break;
-                default:
-                    GtkLayerShell.set_anchor (this,
-                                              GtkLayerShell.Edge.LEFT,
-                                              false);
-                    GtkLayerShell.set_anchor (this,
-                                              GtkLayerShell.Edge.RIGHT,
-                                              true);
-                    break;
-            }
+            GtkLayerShell.set_margin (this,
+                                      GtkLayerShell.Edge.TOP,
+                                      ConfigModel.instance.control_center_margin_top);
+            GtkLayerShell.set_margin (this,
+                                      GtkLayerShell.Edge.BOTTOM,
+                                      ConfigModel.instance.control_center_margin_bottom);
+            GtkLayerShell.set_margin (this,
+                                      GtkLayerShell.Edge.RIGHT,
+                                      ConfigModel.instance.control_center_margin_right);
+            GtkLayerShell.set_margin (this,
+                                      GtkLayerShell.Edge.LEFT,
+                                      ConfigModel.instance.control_center_margin_left);
+
+            // Anchor to north/south edges as needed
+            bool anchorTop = ConfigModel.instance.positionY == PositionY.TOP;
+            GtkLayerShell.set_anchor (this,
+                                      GtkLayerShell.Edge.TOP,
+                                      ConfigModel.instance.fit_to_screen || anchorTop);
+            GtkLayerShell.set_anchor (this,
+                                      GtkLayerShell.Edge.BOTTOM,
+                                      ConfigModel.instance.fit_to_screen || !anchorTop);
+
+            bool anchorLeft = ConfigModel.instance.positionX == PositionX.LEFT;
+            bool anchorRight = ConfigModel.instance.positionX == PositionX.RIGHT;
+            GtkLayerShell.set_anchor (this,
+                                      GtkLayerShell.Edge.RIGHT,
+                                      anchorLeft);
+            GtkLayerShell.set_anchor (this,
+                                      GtkLayerShell.Edge.LEFT,
+                                      anchorRight);
```

Comment:
Removing the `on_map` method results in the X position being flipped. We'd need to switch `anchorLeft` and `anchorRight` to fix this

Reply:
Indeed, that was a small mistake on my end.